### PR TITLE
Add integration tests for text embedding processor/semantic field with remote model

### LINKED ISSUE #1756: [BUG] Add integration tests for k-NN ingestion with ingestion pipeline
### What is the bug?
There is a bug reported in ml-commons that ingestion failed for a knn index with ingestion pipeline. Seems like neural-search is lack of integration tests for such scenarios. Need check and add more integration tests to cover sure both local models and remote connectors.
- issue: https://github.com/opensearch-project/ml-commons/issues/4624

### How can one reproduce the bug?
- the issue was fixed in ml-commons, since that's the root cause. 
- create a knn index
- ingest with ingestion pipeline

### What is the expected behavior?
- add integration tests to cover the scenarios.

### What is your host/environment?
_Operating system, version._

### Do you have any screenshots?
_If applicable, add screenshots to help explain your problem._

### Do you have any additional context?
_Add any other context about the problem._

### PR DESCRIPTION
**Pull request by heemin32**
### Description
This commit adds integration tests for k-NN ingestion with ingestion pipelines and semantic field using remote models via TorchServe connectors.

### Related Issues
Resolves #1756

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/neural-search/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/neuralsearch/ml/SymmetricRemoteModelIT.java [hunk 1]
```diff
@@ -0,0 +1,730 @@
...
+                "description": "Remote symmetric text embedding model",
+                "connector_id": "%s",
+                "model_config": {
+                    "model_type": "TEXT_EMBEDDING",
+                    "embedding_dimension": 128,
+                    "framework_type": "sentence_transformers",
+                    "additional_config": {
+                        "space_type": "l2"
+                    }
+                }
```
Comment (martin-gaievski) [inline src/test/java/org/opensearch/neuralsearch/ml/SymmetricRemoteModelIT.java:689]: isn't it extra closing brace?

File: src/test/resources/remote-models/torchserve/handlers/symmetric_text_embedding_handler.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import torch
+from transformers import AutoTokenizer, AutoModel
+from ts.torch_handler.base_handler import BaseHandler
+
+logger = logging.getLogger(__name__)
+
+class SymmetricTextEmbeddingHandler(BaseHandler):
+    """
+    Simple TorchServe handler for symmetric text embedding using sentence-transformers.
+    This produces 768-dimensional embeddings similar to traced_small_model.
```
Comment (martin-gaievski) [inline src/test/resources/remote-models/torchserve/handlers/symmetric_text_embedding_handler.py]: this should say 128, model `google/bert_uncased_L-2_H-128_A-2` produces embedding with 128-dim

File: src/test/resources/remote-models/torchserve/handlers/symmetric_text_embedding_handler.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        """
+        # Return simple array format for OpenSearch to wrap properly
+        if len(data.shape) == 2:  # Batch of embeddings
+            # Return array of arrays for batch
+            result = [embedding.tolist() for embedding in data]
+        else:  # Single embedding
+            # Return single array
+            result = data.tolist()
+
+        return [result]
```
Comment (vibrantvarun) [inline src/test/resources/remote-models/torchserve/handlers/symmetric_text_embedding_handler.py:138]: Could you remove this extra line.

Comment (heemin32) [reply src/test/resources/remote-models/torchserve/handlers/symmetric_text_embedding_handler.py:138]: This is not an extra line. There is no extra line.

File: src/test/java/org/opensearch/neuralsearch/ml/SymmetricRemoteModelIT.java (hunk 1 above)
Comment (heemin32) [reply src/test/java/org/opensearch/neuralsearch/ml/SymmetricRemoteModelIT.java:689]: Fixed. Thanks.

File: src/test/resources/remote-models/torchserve/handlers/symmetric_text_embedding_handler.py (hunk 2 above)
Comment (heemin32) [reply src/test/resources/remote-models/torchserve/handlers/symmetric_text_embedding_handler.py]: Updated. Thanks.
